recipe/thirdparty/providers/google.go

```diff
@@ -133,6 +134,10 @@ func doGetRequest(req *http.Request) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
+	if resp.StatusCode >= 300 {
+		return nil, errors.New("Provider API did not return a success status code")
```

Comment:
- extract this function to a different file
- Send the status code and body in the error message as well